cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForGenerators.java

```diff
@@ -112,7 +113,7 @@ private boolean setRegulatingControlVoltage(String controlId,
 
         // add qPercent as an extension
         if (!Double.isNaN(qPercent)) {
-            CoordinatedReactiveControl coordinatedReactiveControl = new CoordinatedReactiveControl(gen, qPercent);
+            CoordinatedReactiveControl coordinatedReactiveControl = new CoordinatedReactiveControlImpl(gen, qPercent);
             gen.addExtension(CoordinatedReactiveControl.class, coordinatedReactiveControl);
```

Comment:
I think we should use the new design here to be sure that the extension use the same NetworkFactory (persistent or transient) than the network